src/openassetio-core/include/openassetio/hostApi/Manager.hpp

```diff
@@ -754,6 +753,66 @@ class OPENASSETIO_CORE_EXPORT Manager final {
   void entityExists(const EntityReferences& entityReferences, const ContextConstPtr& context,
                     const ExistsSuccessCallback& successCallback,
                     const BatchElementErrorCallback& errorCallback);
+
+  /**
+   * Callback signature used for a successful entity trait set query.
+   */
+  using EntityTraitsSuccessCallback = std::function<void(std::size_t, trait::TraitSet)>;
+
+  /**
+   * Retrieve the @ref trait_set of one or more @ref entity "entities".
+   *
+   * For example, this may be used to validate that a user-provided
+   * entity reference is appropriate for an operation.
+   *
+   * The trait set returned (via callback) for each @ref
+   * entity_reference varies according to the @p entityTraitsAccess
+   * access mode.
+   *
+   * If @ref access.EntityTraitsAccess.kRead "kRead" is given, the
+   * response will be an exhaustive trait set for the entity. If an
+   * entity does not exist, then the error callback will be invoked
+   * using the @ref errors.BatchElementError.ErrorCode.kEntityResolutionError
+   * "kEntityResolutionError" code.
+   *
+   * If @ref access.EntityTraitsAccess.kWrite "kWrite" is given, the
+   * response will be the minimal trait set required to publish to the
+   * entity reference. If an entity is read-only, the error callback
+   * will be invoked using the @ref errors.BatchElementError.ErrorCode.kEntityAccessError
+   * "kEntityAccessError" code.
+   *
+   * The trait set will include all traits that an entity possesses, not
+   * just those with `resolve`able properties, so that they can be used
+   * to determine the kind of entity.
+   *
+   * An empty trait set is a valid response, for example if the entity
+   * is a new asset with no type constraints.
```

Comment:
[consider] Here, and on the other side, moving to be with the `kWrite` text as I think this only applies there, where as the previous para is for both read/write.